File: output/ChatGPT/avatar/Java/Python/codeforces_629_B.py
n = int(input())
FfriendPerDay = [0] * 367
MfriendPerDay = [0] * 367
answer = 0

for _ in range(n):
    c, a, b = input().split()
    a = int(a)
    b = int(b)
    
    for j in range(a, b+1):
        if c == 'M':
            MfriendPerDay[j] += 1
        else:
            FfriendPerDay[j] += 1
        
        if MfriendPerDay[j] < FfriendPerDay[j]:
            if MfriendPerDay[j] > answer:
                answer = MfriendPerDay[j]
        else:
            if FfriendPerDay[j] > answer:
                answer = FfriendPerDay[j]

print(answer * 2)

File: output/ChatGPT/avatar/Java/C++/codeforces_629_B.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> FfriendPerDay(367, 0);
    vector<int> MfriendPerDay(367, 0);
    int answer = 0;
    for (int i = 0; i < n; i++) {
        char c;
        cin >> c;
        int a, b;
        cin >> a >> b;
        for (int j = a; j <= b; j++) {
            if (c == 'M')
                MfriendPerDay[j]++;
            else
                FfriendPerDay[j]++;
            if (MfriendPerDay[j] < FfriendPerDay[j]) {
                if (MfriendPerDay[j] > answer)
                    answer = MfriendPerDay[j];
            } else {
                if (FfriendPerDay[j] > answer)
                    answer = FfriendPerDay[j];
            }
        }
    }
    cout << answer * 2 << endl;
}